src/query-parser/errors/query-errors.ts

```
@@ -0,0 +1,622 @@
+        `Check if the filter operation is supported for this field type`,
+        `Verify the filter value is of the correct type`,
+        `Consult the documentation for supported filter operations`,
+      ]
+    );
+  }
+}
+
+/**
+ * Type-related errors
+ */
+export abstract class TypeError extends QueryParserError {
```

Biome (1.9.4)

[error] Do not shadow the global "TypeError" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)